includes/data-events/connectors/ga4/class-ga4.php

```diff
@@ -0,0 +1,185 @@
+<?php
+/**
+ * Newspack Data Events Ga4 Connector
+ *
+ * @package Newspack
+ */
+
+namespace Newspack\Data_Events\Connectors;
+
+require_once 'class-event.php';
+
+use Automattic\Jetpack\Device_Detection;
+use Newspack\Data_Events;
+use Newspack\Data_Events\Connectors\GA4\Event;
+use Newspack\Logger;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Main Class.
+ */
+class GA4 {
+
+	/**
+	 * Initialize the class and registers the handlers
+	 *
+	 * @return void
+	 */
+	public static function init() {
+		Data_Events::register_handler( [ __CLASS__, 'handle_reader_logged_in' ], 'reader_logged_in' );
+	}
+
+	/**
+	 * Handler for the reader_logged_in event.
+	 *
+	 * @param int   $timestamp Timestamp of the event.
+	 * @param array $data      Data associated with the event.
+	 * @param int   $client_id ID of the client that triggered the event.
+	 *
+	 * @throws \Exception If the event is invalid.
+	 * @return void
+	 */
+	public static function handle_reader_logged_in( $timestamp, $data, $client_id ) {
+
+		$params = array_merge(
+			self::get_default_params(),
+			[ 'user_id' => $data['user_id'] ]
+		);
+
+		$event = new Event( 'reader_login', $params );
+		self::send_event( $event, $client_id );
+	}
+
+	/**
+	 * Get the default parameters that are added to all events
+	 *
+	 * @return array
+	 */
+	private static function get_default_params() {
+		$params = [
+			'logged_in' => is_user_logged_in() ? 'yes' : 'no',
+		];
+
+		if ( class_exists( 'Automattic\Jetpack\Device_Detection' ) ) {
+			self::log( 'Detecting device using user agent: ' . $_SERVER['HTTP_USER_AGENT'] ?? '(empty)' ); // phpcs:ignore
+			$device                = Device_Detection::is_phone() ? 'phone' : ( Device_Detection::is_tablet() ? 'tablet' : 'desktop' );
+			$params['device_type'] = $device;
+		}
```

Comment:
How does GA treat the event if it's attached to a `_ga` cookie (GA session)? I was hoping all the standard GA tracking would provide all the browser/client data without having to duplicate it ourselves.

Ideally, we would keep events more agnostic and only contain the information related to the action.

Reply:
Even though I'm passing the `client_id` based on the cookie, as the documenation suggests, I don't see anything in the GA dashboard. When I see the event details, they don't have any of these standard dimensions... 

I'll keep looking. I think this is a critical piece of the implementation

Reply:
I've done a long exploration. See pamTN9-5QY-p2#comment-7460

It looks like these params will never be automatically added, but I managed to add a session_id to our events that will allow us to tie them to other front-end events when we process the raw data in BigQuery.

I've updated the testing intructions.